packages/react/src/components/Authenticator/shared/AttributeField.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  ActorContextWithForms,
+  authInputAttributes,
+  getActorContext,
+  translate,
+  isSupportedAuthField,
+} from '@aws-amplify/ui';
+import {
+  PasswordField,
+  PhoneNumberField,
+  TextField,
+} from '../../../primitives';
+import { useAuthenticator } from '../hooks/useAuthenticator';
+
+export function AttributeField({ name, ...passedAttrs }) {
+  const { _state } = useAuthenticator();
+  const { country_code } = getActorContext(_state) as ActorContextWithForms;
+
+  if (!isSupportedAuthField(name)) {
+    console.debug(
+      `Authenticator does not have a default implementation for ${name}. Customize Authenticator.SignUp.FormFields to add your own.`
+    );
+    return null;
+  }
+
+  const defaultAttrs = authInputAttributes[name];
+
+  // merged passed attributes with default attributes
+  const attrs = { ...defaultAttrs, ...passedAttrs } as Record<string, string>;
+
+  // "autocomplete" is "autoComplete" in React:
+  if (!!attrs.autocomplete) {
+    attrs.autoComplete = attrs.autoComplete || attrs.autocomplete;
+    delete attrs.autocomplete;
+  }
+
+  const label = translate<string>(attrs.label);
+  const placeholder = translate<string>(attrs.placeholder);
```

Comment:
Think we can simplify this section of the component (and avoid casting the values of `attrs` to `string`) by changing the structure of the passed props in to the different Field components, e.g.

```
<PhoneNumberField
  {...defaultProps}
  autoComplete={defaultProps.autocomplete || defaultProps.autoComplete}
  defaultCountryCode={country_code}
  isRequired
  label={label}
  labelHidden
  name="phone"
  placeholder={placeholder}
/>
```

What do you think?